internal/app/subsystems/api/http/promise.go

```diff
@@ -122,17 +121,11 @@ func (s *server) completePromise(c *gin.Context) {
 		err  error
 	)
 
-	switch strings.ToUpper(body.State) {
-	case promise.Resolved.String():
-		resp, err = s.service.ResolvePromise(id, &header, body)
-	case promise.Rejected.String():
-		resp, err = s.service.RejectPromise(id, &header, body)
-	case promise.Canceled.String():
-		resp, err = s.service.CancelPromise(id, &header, body)
-	default:
+	if !body.State.In(promise.Resolved | promise.Rejected | promise.Canceled) {
 		c.JSON(http.StatusBadRequest, api.HandleValidationError(errors.New("invalid state")))
```

Comment:
We just need a return after this line, sorry I didn't see that earlier

Reply:
I added the return after this line, thanks!